wrangler-core/src/main/java/io/cdap/wrangler/executor/RecipePipelineExecutor.java

```diff
@@ -161,4 +186,38 @@ private List<Directive> getDirectives() throws RecipeException {
     }
     return directives;
   }
+
+  private Schema generateOutputSchema(Schema inputSchema, List<Row> output) throws RecordConvertorException {
+    Set<Field> outputFields = new HashSet<>();
+    for (Row row : output) {
+      for (Pair<String, Object> rowField : row.getFields()) {
+        String fieldName = rowField.getFirst();
+        Object fieldValue = rowField.getSecond();
+
+        Schema existing = inputSchema.getField(fieldName) != null ? inputSchema.getField(fieldName).getSchema() : null;
+        Schema generated = fieldValue != null && (existing == null || !isValidSchemaForValue(existing, fieldValue)) ?
+          generator.getSchema(fieldValue, fieldName) : null;
+
+        if (generated != null) {
+          outputFields.add(Schema.Field.of(fieldName, generated));
+        } else if (existing != null) {
+          outputFields.add(Schema.Field.of(fieldName, existing));
```

Comment:
Do you want to add a field schema for each row?

Reply:
Could you clarify what exactly the issue here is?